packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -370,6 +429,36 @@ void main() {
     expect(find.text('ddd1'), findsOneWidget);
   });
 
+  testWidgets('Holding scroll and Scroll pointer signal will update ScrollDirection.forward / ScrollDirection.reverse', (WidgetTester tester) async {
+    ScrollDirection lastUserScrollingDirection;
+
+    final ScrollController controller = ScrollController();
+    await tester.pumpWidget(buildTest(controller: controller));
+
+    final Offset scrollEventLocation = tester.getCenter(find.byType(NestedScrollView));
+    final TestPointer testPointer = TestPointer(1, ui.PointerDeviceKind.mouse);
+    // Create a hover event so that |testPointer| has a location when generating the scroll.
+    testPointer.hover(scrollEventLocation);
+
+    controller.addListener(() {
+      if(controller.position.userScrollDirection != ScrollDirection.idle)
+        lastUserScrollingDirection = controller.position.userScrollDirection;
+    });
+
+    await tester.drag(find.byType(NestedScrollView), const Offset(0.0, -20.0), touchSlopY: 0.0);
+
+    expect(lastUserScrollingDirection, ScrollDirection.reverse);
+
+    await tester.sendEventToBinding(testPointer.scroll(const Offset(0.0, 20.0)));
```

Comment:
Because I want to make sure that the mouse wheel logic doesn't have a different effect than the other sliding logic (except for acceleration and overscroll)

Now that doesn't seem like a very good practice, I will rewrite the correct test.